utils/timeout.h

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2019 BiiLabs Co., Ltd. and Contributors
+ * All Rights Reserved.
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the MIT license. A copy of the license can be found in the file
+ * "LICENSE" at the root of this distribution.
+ */
+
+#ifndef UTILS_CTIMER_H_
+#define UTILS_CTIMER_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @file timeout.h
+ * @brief Implementation of function timer
+ */
+
+#include <pthread.h>
+#include <signal.h>
+#include <stdio.h>
+#include <sys/time.h>
+#include "accelerator/errors.h"
+
+/**
+ * Helper function to set timer for a given callback
+ * @param[in] timer Timeout to wait until callback being halted
+ * @param[in] callback The actual function being called
+ * @param[in] args The arguments to pass to the callback function
+ *
+ * @return
+ * - PTHREAD_CANCELED on time limit exceeded
+ * - others depending on the callback function
+ */
+status_t timeout_join(const struct itimerval *timer, void *callback, void *args);
```

Comment:
According to fork-join pattern, there should be something like "fork" once you provide "join".